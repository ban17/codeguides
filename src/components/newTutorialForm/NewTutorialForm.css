.new-tutorial-form {
  width: 55rem;
  margin: 1rem auto;
  text-align: left;

  label {
    display: inline-block;
    width: 15rem;
  }
}
