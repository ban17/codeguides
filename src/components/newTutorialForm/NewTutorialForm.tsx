import './NewTutorialForm.css'

import axios from 'axios'
import React, { useState } from 'react'

import InputField from '@/components/inputField/InputField'

const NewTutorialForm: React.FC = () => {
  // TODO use one state for all input values somehow xD
  const [name, changeName] = useState('')
  const [url, changeUrl] = useState('')
  const [tags, changeTags] = useState('')
  const [category, changeCategory] = useState('')

  const handleSubmit = (e: any) => {
    e.preventDefault()

    const data = {
      tags,
      url,
      'tutorial-name': name,
      'main-category': category
    }

    axios({
      method: 'post',
      url: 'http://localhost:3007/api/tutorials/add',
      data,
      headers: { 'Content-Type': 'application/json' }
    })
      .then(response => {
        // handle success
        // eslint-disable-next-line
        console.log(response)
      })
      // eslint-disable-next-line handle-callback-err
      .catch(error => {
        // handle error
        // eslint-disable-next-line
        console.log(error)
      })
  }

  return (
    <form
      action="http://localhost:3007/api/tutorials/add"
      method="POST"
      className="new-tutorial-form"
      onSubmit={handleSubmit}
    >
      <InputField
        name="tutorial-name"
        label="Tutorial name:"
        handleOnChange={changeName}
        value={name}
      />
      <InputField
        name="url"
        label="Git repo:"
        handleOnChange={changeUrl}
        value={url}
      />
      <InputField
        name="image"
        label="Image:"
        handleOnChange={() => {}}
        type="file"
        accept="image/*"
      />
      <InputField
        name="tags"
        label="Tags:"
        handleOnChange={changeTags}
        value={tags}
      />
      <InputField
        name="main-category"
        label="Main category:"
        handleOnChange={changeCategory}
        value={category}
      />
      <input type="submit" value="Send" />
    </form>
  )
}

export default NewTutorialForm
