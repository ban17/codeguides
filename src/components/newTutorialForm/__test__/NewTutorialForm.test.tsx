import { shallow } from 'enzyme'
import React from 'react'

import NewTutorialForm from '@/components/newTutorialForm/NewTutorialForm'

describe('NewTutorialFrom', () => {
  it('should be rendered correctly', () => {
    expect(shallow(<NewTutorialForm />)).toMatchSnapshot()
  })
})
