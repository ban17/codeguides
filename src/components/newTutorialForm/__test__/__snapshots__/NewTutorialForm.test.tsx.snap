// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTutorialFrom should be rendered correctly 1`] = `
<form
  action="http://localhost:3007/api/tutorials/add"
  className="new-tutorial-form"
  method="POST"
  onSubmit={[Function]}
>
  <InputField
    handleOnChange={[Function]}
    label="Tutorial name:"
    name="tutorial-name"
    value=""
  />
  <InputField
    handleOnChange={[Function]}
    label="Git repo:"
    name="url"
    value=""
  />
  <InputField
    accept="image/*"
    handleOnChange={[Function]}
    label="Image:"
    name="image"
    type="file"
  />
  <InputField
    handleOnChange={[Function]}
    label="Tags:"
    name="tags"
    value=""
  />
  <InputField
    handleOnChange={[Function]}
    label="Main category:"
    name="main-category"
    value=""
  />
  <input
    type="submit"
    value="Send"
  />
</form>
`;
